aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -18,29 +28,89 @@ pub enum AggregatorReadMode {
 /// Returns a value of an aggregator from cache or global storage.
 ///   - Ok(..)       if aggregator value exists
 ///   - Err(..)      otherwise.
-pub trait AggregatorResolver {
+pub trait TAggregatorView {
+    type Identifier;
+
+    fn get_aggregator_v1_state_value(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<StateValue>>;
+
+    fn get_aggregator_v1_value(&self, id: &Self::Identifier) -> anyhow::Result<u128> {
+        let maybe_state_value = self.get_aggregator_v1_state_value(id)?;
+        // TODO: consider reviving Option<u128>?
+        let u128_bytes = maybe_state_value
+            .expect("Aggregator V1 cannot be deleted")
+            .into_bytes();
+        bcs::from_bytes(&u128_bytes)
+            .map_err(|_| anyhow::Error::msg("Failed to deserialize aggregator value to u128"))
+    }
+
+    // TODO: Nested options are used due to cyclic dependency. Aggregator crate needs refactoring.
+    fn get_aggregator_v1_state_value_metadata(
+        &self,
```

Comment:
What's the metadata for aggregator state value? Where do we need to use it?

Reply:
We use the metadata to calculate refunds for removed /. changed state items, and since V1 is a state item, I had to add an explicit API for that (we do not have a single StateView)

Reply:
Added a comment